docs/tutorial/03-let-the-snake-move.md

```diff
@@ -80,44 +70,64 @@ defp draw_objects(graph, %{snake: %{body: body}}) do
 end
 ```
 
-There's one last piece of information missing before we start working on making the snake move. We need to add the snake direction to the state, so that we know how to move it. Later on, we'll be able to change its direction, but for now it will only move to the right.
+There's one last piece of information missing before we can start working on making the snake move. We need to add the snake direction to the state, so we know how to update its body coordinates. Later on, we'll be able to change its direction, but for now it will only move to the right.
+
+So how can we represent the snake direction? Since this is a 2 dimensional game, it seems natural to represent it as a pair of values, one for each axis (x, y). Or, translating it to the Elixir world, as a tuple `{x, y}`.
+```
+          │-1
+          │
+          │
+          │
+-1 ───────┼────────> 1
+          │0
+          │
+          │
+          │1
+```
+
+Based on the Cartesian system above, we can define the direction of the snake as:
+- left: `{-1, 0}`
+- right: `{1, 0}`
+- top: `{0, -1}`
+- down: `{0, 1}`
+
+Let's add this information to the snake object:
 
 ```elixir
 state = %{
   width: number_of_columns,
   height: number_of_rows,
-  snake: %{body: [{9, 9}, {10, 9}, {11, 9}], size: 5, direction: {1, 0}}
+  snake: %{body: [{11, 9}, {10, 9}, {9, 9}], direction: {1, 0}}
 }
 ```
 
-The snake movement is described by a shift of its body coordinates. Looking at the list that represents the snake body, we can see that this is equivalent to adding a new pair of coordinates to the beginning of the list and removing another from the end, so that the snake size is preserved.
-
-Now that we have our snake movement logic figured out, let's go ahead and implement a `move_snake/1` function.
+Now we're ready to implement the movement logic. Let's recall what happens during the game when you turn the snake. At first, only the head turns, right? It's the head that controls the movement. At each step, the snake moves as if we were placing a new head on the tile that we want to move to and removing the last tile in the snake tail, so that its size is preserved. Or, in mathematical terms, that would be equivalent to adding a new tuple to the beginning of the list and removing another one from the end.
 
 ```elixir
 defp move_snake(%{snake: snake} = state) do
-  %{body: body, size: size, direction: direction} = snake
+  %{body: body, direction: direction} = snake
```

Comment:
Was wondering yesterday: should move this information to the root level of the `state` for easier access? Like so:
```elixir
state = %{
  direction: {1, 0},
  snake: [{11, 9}, {10, 9}, {9, 9}],
  # etc.
}
```
🤔

Reply:
Not sure I understand :thinking:

Reply:
Sorry 🙈 

I just meant, we could also structure the `state` differently, so that we don't have to dig too deep to access certain keys.

For instance:
```elixir
defp move_snake(%{snake: snake, direction: direction} = state) do
  # ... snake would be equal to the snake body
```

Just a random thought I had when reading over it.

Reply:
Oh, but later on we're going to add the pellet to the state and it could become ambiguous (?) what the direction refers to.

Reply:
Valid point. Let's just sick with the current approach and see how it turns out in the end. We can keep it in the back of our head 🧠